### PR TITLE
feat(project-funds): add initial version of project funds

### PR DESCRIPTION
**Pull request by ovflowd**
This PR introduces an initial version of the "Project Initiatives Funds".

Closes https://github.com/openjs-foundation/cross-project-council/issues/953.

### CODE REVIEW

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [inline project-resources/PROJECT_FUNDS.md:24]: It would be interesting to have a range of what is feasible to be asked. And if different project categories (At-large, Impact, Incubation) should have different limits.

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [inline project-resources/PROJECT_FUNDS.md:19]: It would be interesting to know if CPC voting must be unilateral, or how exactly the voting works.

I would also suggest for us to have a Markdown file (Basically a table) like we have for Travel Funds to keep documented previous "approved project funds requests".

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [inline project-resources/PROJECT_FUNDS.md:7]: > and give visibility/recognition to the champions of these project-specific initiatives, highligthing the individuals behind them.

I don't think that should be the purpose of this fund. I would suggest simplifying that section (or anchoring back to principles that aren't subject to a controversy.).

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [inline project-resources/PROJECT_FUNDS.md:13]: FWIW, I think offering a template makes everyone's life easier, so I think we just should do that.

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [inline project-resources/PROJECT_FUNDS.md:3]:
```suggestion
The OpenJS Foundation (OpenJSF) provides a set of monetary funds for projects of the OpenJSF. 
```

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [inline project-resources/PROJECT_FUNDS.md:19]:
```suggestion
The requests should go through the OpenJSF Cross Project Council (CPC) and must have approval from the CPC members. 
```

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:7]: Hmm, that's what I've understood from what @mcollina wrote on the original issue. Maybe bad phrasing from my side?

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [inline project-resources/PROJECT_FUNDS.md:24]:
```suggestion
The level of funding level will be determined on case per case basis after consideration of the request and discussion with the project maintainers.
```

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:13]: Makes sense. WDYT the template should contain?

File: project-resources/PROJECT_FUNDS.md
Comment (mcollina) [reply project-resources/PROJECT_FUNDS.md:7]: I don't think I ever said that.
In fact, it's the other way - funds will be needed where our volunteer-based model fails.

Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:7]: @mcollina it is from this paragraph that I took what I wrote " Volunteers are driven by their motives but usually require visibility/recognition of their work. However, this model fails for tasks without recognition: if you do your job well, no one would know you did it."

I might have misunderstood or just poorly phrased my statements. Could you suggest an alternative patch here for the PR? Thank you!

File: project-resources/PROJECT_FUNDS.md
Comment (erickzhao) [reply project-resources/PROJECT_FUNDS.md]: Do we have existing approval/voting processes we can fall back on?

File: project-resources/PROJECT_FUNDS.md
Comment (erickzhao) [reply project-resources/PROJECT_FUNDS.md]: > And if different project categories (At-large, Impact, Incubation) should have different limits.

Would a note clarifying that project status will affect the allocated budget help?

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md]: The only similar process I'm aware of is the travel fund...

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [reply project-resources/PROJECT_FUNDS.md:7]: Regardless, I don't think there's a benefit in attaching a value statement of that nature to the disbursement of funds.

Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:7]: I'm fine with any path, just tried to give an introductory text about what the project funds are and why they exist. Sounds like it was not a good text, if anybody has a better suggestion please comment, I'd love to have this aligned with what we believe that fits better here 🙃

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [reply project-resources/PROJECT_FUNDS.md]: > It would be interesting to know if CPC voting must be unilateral, or how exactly the voting works.

_"The request must be made to the CPC for evaluation. The CPC evaluates the request following its [decision-making process](https://github.com/openjs-foundation/cross-project-council/blob/a49a8d2e04716525116c0ef787e1187d4dcd9e3e/CPC-CHARTER.md#section-9-decision-making) and makes a recommendation to the Board."_

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [reply project-resources/PROJECT_FUNDS.md]: > Would a note clarifying that project status will affect the allocated budget help?

I think that's not necessarily true and also not super useful.

Comment (tobie) [reply project-resources/PROJECT_FUNDS.md]: I don't think we have a good answer for this right now. :(

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [inline project-resources/PROJECT_FUNDS.md:19]:
```suggestion
The request must be made to the CPC for evaluation. The CPC evaluates the request following its [decision-making process](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-9-decision-making) and makes a recommendation to the Board.
```

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:13]: Which points, the ones @joesepi wrote?

Comment (tobie) [reply project-resources/PROJECT_FUNDS.md:13]: My understanding is @erickzhao was referring those in the quoted sentence:

> To access the funds, the project would need to prepare a 1-pager plan, including the purpose and type fo activity, success criteria, timeline, and funds requested

File: project-resources/PROJECT_FUNDS.md
Comment (mcollina) [inline project-resources/PROJECT_FUNDS.md:7]:
```suggestion
Alongside these lines, the fund exists to support the development of the project when the volunteer-based approach is not enough to support the community, e.g. infrastructure, tools, or security.
```

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [reply project-resources/PROJECT_FUNDS.md:7]: Didn't mean to be negative about your input. Apologies if it came across that way. I'm very thankful that you got this started.

Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md]: Oh no, it's fine, I just honestly don't know what we should replace the current entry with 👀

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:12]: Oh my goodness, how I missed that! Sorry @erickzhao 🙈

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md]: Thank you for the patch, @mcollina!

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md]: I believe Matteo already applied a patch here 🎉

**Review by joesepi — APPROVED** [review]

**Review by ljharb — APPROVED** [review]

File: project-resources/PROJECT_FUNDS.md
Comment (tobie) [reply project-resources/PROJECT_FUNDS.md:12]: There seemed to be agreement to have a template for this. Can we bake it into this PR?

Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:12]: Sure thing!

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ovflowd** [comment]
@openjs-foundation/cpc are we good to merge this?

File: project-resources/PROJECT_FUNDS.md
Comment (timmywil) [inline project-resources/PROJECT_FUNDS.md:10]:
```suggestion
### How might a project request funds?
```

File: project-resources/PROJECT_FUNDS.md
Comment (timmywil) [inline project-resources/PROJECT_FUNDS.md:59]:
```suggestion
The level of funding level will be determined on case by case basis after consideration of the request and discussion with the project maintainers.
```

**Review by timmywil — APPROVED** [review]
Approved pending a couple minor grammatical suggestions

**Review by eemeli — CHANGES_REQUESTED** [review]
I fully approve of the general shape of this, but the practical bits need a bit more work. Templates work well for issues, but they're clumsy for PRs. Even more so, a PR needs to effect some change.

We already have a similar process in place for [travel funds](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/project-resources/MEMBER_TRAVEL_FUND.md#requesting-travel-funds), we should follow that precedent here.

File: project-resources/PROJECT_FUNDS.md
Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:4]: > Templates work well for issues, but they're clumsy for PRs. Even more so, a PR needs to effect some change.

I wonder if requests should be opened as an issue then? The idea behind a PR is so that either:

- A very visible way of voting against something
- The file change would maybe be a document we use for "logging/auditing" past approved funds, similar to how we do with expense requests for travel funds.

I'm open for suggestions of course :)

Comment (eemeli) [reply project-resources/PROJECT_FUNDS.md:4]: PR is better than issue, because it allows for tracking approval much better and leaves an explicit trace in the repo when merged.

As I mention in my previous comment, my suggestion would be to imitate our pre-existing process on travel funds.

Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md:4]: Hmm, so we are on the same page, you'd like me to add to this PR a Markdown file with a table similar to the one from the Travel Expenses one?

Just trying to understand what are the next steps here :)

Comment (eemeli) [reply project-resources/PROJECT_FUNDS.md:4]: Yes. We already have one process for asking for funds, and it works. Let's make sure that this process matches that one, rather than creating a different process.

The instructions for requesting travel funds have stood the test of time rather well, let's effectively reuse them here.

Comment (ovflowd) [reply project-resources/PROJECT_FUNDS.md]: Hey, @eemeli, I've just pushed the changes! I apologise for the delay.

File: PROJECT_FUND/2023.md
Comment (eemeli) [inline PROJECT_FUND/2023.md:4]:
```suggestion
First Name | Last Name | Project | Amount requested | Pull Request date | Pull Request link
-|-|-|-|-|-
```

The original issue #953 suggested that when approved, funds would be transferred to a crowdfunding platform from which the project itself may manage their further distribution. This practice is also implied by the "Distribution of the Funds" section of the PR template.

If that is still the plan, the expense report references here are out of place.

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:56]: This post-mortem section probably doesn't belong in the template? As its own comment states, it should be ad-hoc, and written afterwards. If these sections are included in the PR's description, adding their contents later will not provide any notification to the PR's listeners.

It might be more appropriate to recommend or require adding a comment on the PR later with this information.

**Review by eemeli — CHANGES_REQUESTED** [review]
Much better! A couple of small things still need adjustment, see line comments.

File: PROJECT_FUND/2023.md
Comment (ovflowd) [reply PROJECT_FUND/2023.md:4]: Gotcha, I have forgotten about that. Should we add a field for linking to the crowdfunding platform? (Like basically a link for the campaign?) I bet it would be something like OpenCollective?

**Comment by ovflowd** [comment]
Uupise, @ljharb accidentally re-requested a review from you 😅

**Review by ljharb — APPROVED** [review]
still on board :-p

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:16]:
```suggestion
## Requesting Project Funds
```

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:59]:
```suggestion
## Process
```

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:20]:
```suggestion
Funds should be requested on an ad-hoc basis through a PR requesting the funds.
The level of funding will be determined on case by case basis
after consideration of the request and discussion with the project maintainers.

The requests should follow a specific template,
and contain relevant information regarding the project initiative,
as well as any information that might be useful to highlight the need of funds and their benefits.
```

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:66]:
```suggestion
```

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:68]:
```suggestion
## Evaluation
```

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:70]:
```suggestion
An evaluation should be done after the conclusion of the timeline of the fund request. Ideally as a comment on the same PR as it would notify the current people subscribed to the Pull Request. The template below contains the recommended format for the evaluation, but we believe more/less information can be added by the champions of the request through good judgement.
```

It might be best to avoid such literally morbid terminology.

File: project-resources/PROJECT_FUND.md
Comment (eemeli) [inline project-resources/PROJECT_FUND.md:73]:
```suggestion
## Post-request Evaluation
```

**Review by eemeli — APPROVED** [review]
"FAQ" is a bit of a misnomer for the bulk of the instructions, and a few other rewordings. But take these as recommendations, I'm happy with how this is now looking.

**Comment by ovflowd** [comment]
Merging this as all code-review changes were applied! Thank you all for your contribution/collaboration with this PR!

**Comment by ovflowd** [comment]
For some reason, Linkinator hang. So I'm doing a force merge.

**Pull request merged**
